### PR TITLE
Feat/share-data: create tags field component

### PR DESCRIPTION
**Pull request by 0xLodz**
### Behavior
This PR add the TagsField component, allowing the user to add and remove tags. This component is needed for the [share data step 3](https://github.com/okp4/dataverse-portal/issues/247).

### Screenshot
image

### CODE REVIEW

File: src/ui/view/share/data/i18n/share_data_de.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "type-press-enter": "Tippen Sie und drücken Sie die <span>Eingabetaste</span>"
```
Comment (ccamel) [inline src/ui/view/share/data/i18n/share_data_de.json:2]: I don't really like having HTML tags in translations. It's something we can discuss, but the idea of translations is to have something purely textual, even if the [library we use](https://react.i18next.com/latest/trans-component) allows for tag interpolation.

Comment (0xLodz) [reply src/ui/view/share/data/i18n/share_data_de.json:2]: We probably can split this translation in two separated translation, even though the first part of the translation left alone doesn't have that much sense. Using tags for translation is a common practice used in several libraries to isolate a part of it's content for styling. When used regularly (e.g. with `t` from `useTranslation`), it won't create any tag.

File: src/ui/view/share/data/tagsField/tagsField.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setValue(event.target.value)
+    },
+    [setValue]
+  )
+
+  const handleKeyDown = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Enter') {
```
Comment (oussama-aitbaziz-dev) [inline src/ui/view/share/data/tagsField/tagsField.tsx:27]: This might not work in some older browsers we also need to handle `event.keyCode === 13`

**Review by fredericvilcot — APPROVED** [review]
LGTM!

File: src/ui/view/share/data/tagsField/tagsField.tsx (hunk 2 above)
Comment (0xLodz) [reply src/ui/view/share/data/tagsField/tagsField.tsx:27]: Which browser are you thinking of? I think the support is good for the required browsers. https://caniuse.com/?search=event.key

Comment (oussama-aitbaziz-dev) [reply src/ui/view/share/data/tagsField/tagsField.tsx:27]: Im used to handling key codes and key to make sure everything works everywhere :)

File: src/ui/view/share/data/tagsField/tagsField.scss [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    .okp4-dataverse-portal-tag-main {
+      min-height: $tag-height;
+    }
+
+    .okp4-dataverse-portal-tags-field {
+      position: relative;
+      height: $tag-height;
+
+      &-placeholder {
```
Comment (lolottetheclash) [inline src/ui/view/share/data/tagsField/tagsField.scss:27]: This way to handle classnames works fine, but to stay in line with the way we manage classnames in general, can you keep our original way of doing things?

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
Nice job! It's looking great!
Some minor changes, and one to keep in mind for v2:
currently, the way to manage tags is based on the value, which is a string.
The problem is that we can add the same tag twice, and when we want to delete one of them, both disappear.
It will be interesting (for V2) to check with the product team whether it's allowed to add the same tag twice, and if not, how to signal to the user that this isn't allowed. 
If it is allowed, we'll need to think about modifying the tag data format to add an id, for example.

**Review by lolottetheclash — CHANGES_REQUESTED** [review]

File: src/ui/view/share/data/tagsField/tagsField.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        const tag = event.currentTarget.value.trim()
```
[the remark this answers is not in the document]
Comment (0xLodz) [reply src/ui/view/share/data/tagsField/tagsField.tsx]: I agree that it improves code readability. The code was updated with a renamed value state `inputValue`.

File: src/ui/view/share/data/i18n/share_data_de.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xLodz) [reply src/ui/view/share/data/i18n/share_data_de.json]: Oops. Thanks!

File: src/ui/view/share/data/tagsField/tagsField.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      ))}
+      <div className="okp4-dataverse-portal-tags-field">
+        <span
+          className={classNames('okp4-dataverse-portal-tags-field-placeholder', { hidden: value })}
+        >
+          <Trans
+            components={{
+              span: <span className="okp4-dataverse-portal-tags-field-placeholder-action" />
+            }}
+            i18nKey="shareData:type-press-enter"
```
[the remark this answers is not in the document]
Comment (0xLodz) [reply src/ui/view/share/data/tagsField/tagsField.tsx]: Good catch. Fixed in [7659241](https://github.com/okp4/dataverse-portal/pull/390/commits/76592413368f55b72abfc9c6d6e175648d36a68b)!

**Comment by 0xLodz** [comment]
> Nice job! It's looking great! Some minor changes, and one to keep in mind for v2: currently, the way to manage tags is based on the value, which is a string. The problem is that we can add the same tag twice, and when we want to delete one of them, both disappear. It will be interesting (for V2) to check with the product team whether it's allowed to add the same tag twice, and if not, how to signal to the user that this isn't allowed. If it is allowed, we'll need to think about modifying the tag data format to add an id, for example.

Thanks for your feedback, we will have a discussion with the product team for your suggested improvements.
I think similar tags that disappear can be handled by the parent.
But yes, adding the same tag twice is the responsibility of this component. 👍

File: src/ui/view/share/data/i18n/share_data_de.json (hunk 1 above)
Comment (0xLodz) [reply src/ui/view/share/data/i18n/share_data_de.json]: We finally split the translation and removed the tag.

**Review by lolottetheclash — APPROVED** [review]
Great job, thanks !

**Pull request merged**